src/tests/performance/ping_pong.rs

```diff
@@ -114,95 +123,85 @@ async fn throughput() {
-
-        for _ in 0..peers {
-            synth_handles.push(tokio::spawn(async move {
-                // Create a synthetic node, enable handshaking and auto-reply
-                let mut synth_node = SyntheticNode::builder()
-                    .with_full_handshake()
-                    .with_all_auto_reply()
-                    .build()
-                    .await
-                    .unwrap();
-                synth_node.connect(node_addr).await.unwrap();
-
-                for _ in 0..PINGS {
-                    let nonce = Nonce::default();
-                    let expected = Message::Pong(nonce);
-
-                    // send Ping(nonce)
-                    synth_node
-                        .send_direct_message(node_addr, Message::Ping(nonce))
-                        .await
-                        .unwrap();
-
-                    let now = tokio::time::Instant::now();
-                    loop {
-                        match synth_node.recv_message_timeout(PING_TIMEOUT).await {
-                            Err(err) if err.kind() == std::io::ErrorKind::TimedOut => {
-                                metrics::histogram!(METRIC_NAME, duration_as_ms(PING_TIMEOUT))
-                            }
-                            Ok((_, message)) if message == expected => {
-                                metrics::histogram!(METRIC_NAME, duration_as_ms(now.elapsed()))
-                            }
-                            // If the nonce doesn't match then we treat it as a response to an already timed out Ping
-                            // (which has already been handled, so we skip it).
-                            Ok((_, Message::Pong(_))) => continue,
-                            result => {
-                                panic!("Failed to receive {:?}, got {:?}", expected, result)
-                            }
-                        }
-
-                        break;
-                    }
-                }
-
-                synth_node.shut_down();
-            }));
+        for _ in 0..synth_count {
+            synth_handles.push(tokio::spawn(simulate_peer(node_addr)));
         }
 
         // wait for peers to complete
         for handle in synth_handles {
-            handle.await.unwrap();
+            let _ = handle.await;
         }
 
         let time_taken_secs = test_start.elapsed().as_secs_f64();
 
-        // grab latencies from metrics recoder
+        // get latency stats
         let latencies = simple_metrics::histograms()
             .lock()
-            .get(&metrics::Key::from_name(METRIC_NAME))
+            .get(&metrics::Key::from_name(METRIC_LATENCY))
             .unwrap()
             .value
             .clone();
 
         // add stats to table display
         table.add_row(RequestStats::new(
-            peers as u16,
+            synth_count as u16,
             PINGS,
             latencies,
             time_taken_secs,
         ));
+
+        println!("{}", table);
```

Comment:
Suggestion: only print the table at the end of the run?

Reply:
Debug aid left behind :D will remove.